pkg/tsdb/zipkin/client.go

```diff
@@ -0,0 +1,38 @@
+package zipkin
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+)
+
+type ZipkinClient struct {
+	logger     log.Logger
+	url        string
+	httpClient *http.Client
+}
+
+func New(url string, hc *http.Client, logger log.Logger) (ZipkinClient, error) {
+	client := ZipkinClient{
+		logger:     logger,
+		url:        url,
+		httpClient: hc,
+	}
+	return client, nil
+}
+
+// Services returns list of services
+// https://zipkin.io/zipkin-api/#/default/get_services
+func (z *ZipkinClient) Services() ([]string, error) {
+	services := []string{}
+	res, err := z.httpClient.Get(fmt.Sprintf("%s/api/v2/services", z.url))
```

Comment:
In case `z.url` has a trailing slash already, it's probably safer to use https://pkg.go.dev/net/url#JoinPath here

Reply:
Fixed in https://github.com/grafana/grafana/pull/96031/commits/02c873ab0e117067c7bc3d0c6bef47ca433b0490